src/acme/ui/CustomerListUI.java

```diff
@@ -0,0 +1,353 @@
+import java.awt.Component;
+import java.awt.Font;
+import java.awt.Panel;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.sql.ResultSet;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Set;
+import java.util.UUID;
+
+import javax.swing.DefaultCellEditor;
+import javax.swing.JButton;
+import javax.swing.JLabel;
+import javax.swing.JOptionPane;
+import javax.swing.JScrollPane;
+import javax.swing.JTable;
+import javax.swing.JTextField;
+import javax.swing.SwingConstants;
+import javax.swing.table.TableCellEditor;
+import javax.swing.table.TableCellRenderer;
+import javax.swing.table.TableColumn;
+
+
+
+import acme.pd.User;
+import acme.pd.Customer;
+import acme.pd.MapIntersection;
+import acme.pd.Road;
+
+public class CustomerListUI extends AcmeBaseJPanel {
+  Customer [] customersArray;
+  
+  public CustomerListUI() {
+    super();
+  }
+  
+  public void buildPanel() {
+  
+    setLayout(null);
+    
+    Color lightBlue = new Color(93, 184, 202);
+    
+    //title label
+    JLabel lblCustomers = new JLabel("Customers");
+    lblCustomers.setFont(new Font("Lucida Grande", Font.PLAIN, 24));
+    lblCustomers.setBounds(20, 20, 200, 27);
+    add(lblCustomers);
+    
+    //my test data
+    HashMap<UUID, Customer> testData = getCompany().getCustomer();
+    Collection<Customer> values = testData.values();
+    ArrayList<Customer> listOfValues = new ArrayList<Customer>(values);
+    Set<UUID> keySet = testData.keySet();
+    ArrayList<UUID> listOfKeys = new ArrayList<UUID>(keySet);
+    customersArray = listOfValues.toArray(new Customer[listOfValues.size()]);
+    Object[][] objects = getObjectsForTable(customersArray);
+    String[] cols = new String [] {
+        "Number", "Name", "Location", "Active", "Edit"
+    };
+    
+    
+
+    //scroll and table. I made this in netbeans, so it looks a little different
+    JScrollPane jScrollPane1;
+    JTable jTable1 = new JTable();
+
+    //set up table
+    jTable1.setBorder(javax.swing.BorderFactory.createEtchedBorder());
+    jTable1.setFont(new java.awt.Font("Lucida Grande", 0, 16)); 
+    jTable1.setModel(new javax.swing.table.DefaultTableModel(objects, cols) {
+        Class[] types = new Class [] {
+            java.lang.String.class, java.lang.String.class, java.lang.String.class, java.lang.Boolean.class, javax.swing.JButton.class
+        };
+        boolean[] canEdit = new boolean [] {
+            //need last one true so can click button
+            false, false, false, false, true
+        };
+
+        public Class getColumnClass(int columnIndex) {
+            return types [columnIndex];
+        }
+
+        public boolean isCellEditable(int rowIndex, int columnIndex) {
+            return canEdit [columnIndex];
+        }
+    });
+    
+    //needed for buttons in table
+    TableColumn editCol = jTable1.getColumnModel().getColumn(4);
+    JButton editButton = new JButton("edit");
```

Comment:
Is this editButton ever used? if not, we can remove it (just general cleanup)

Reply:
Took it out